github/src/webhook.rs

```diff
@@ -15,34 +15,42 @@ pub const DELIVERY_ID_HEADER: &str = "X-Github-Delivery";
 ///
 /// Github API docs: https://developer.github.com/webhooks/#delivery-headers
 pub const SIGNATURE_HEADER: &str = "X-Hub-Signature";
+pub const SIGNATURE_256_HEADER: &str = "X-Hub-Signature-256";
 
 #[derive(Clone, Debug)]
 pub struct Webhook {
     pub event_type: EventType,
     pub delivery_id: String,
     pub signature: Option<String>,
+    pub signature_256: Option<String>,
     pub body: Vec<u8>,
 }
 
 impl Webhook {
     pub fn check_signature(&self, key: Option<&[u8]>) -> bool {
-        match (key, &self.signature) {
-            (Some(key), Some(signature)) if signature.starts_with("sha1=") => {
```

Comment:
The check that the signatures start with "sha1=" or "sha256=" should probably be kept as a safe guard. The new code seems to just assume this is true without the invariant ever being checked.

Reply:
GitHub's documentation has told that sha1 is only kept for backwards compatibility.  But the main docs have completely replaced them with sha256.  If it doesn't have it, then you lose the protection of validated web hooks (and someone could just send you garbage)